src/libaktualizr/primary/aktualizr.cc

```diff
@@ -170,13 +170,30 @@ boost::signals2::connection Aktualizr::SetSignalHandler(
   return sig_->connect(handler);
 }
 
-std::unique_ptr<StorageTargetRHandle> Aktualizr::GetStoredTarget(const Uptane::Target &target) {
+std::vector<Uptane::Target> Aktualizr::GetStoredTargets() {
+  try {
+    return storage_->getTargetFiles();
+  } catch (...) {
+    return {};
+  }
+}
+
+bool Aktualizr::DeleteStoredTarget(const Uptane::Target &target) {
+  try {
+    storage_->removeTargetFile(target.filename());
+  } catch (...) {
+    return false;
+  }
+  return true;
+}
+
+std::unique_ptr<StorageTargetRHandle> Aktualizr::OpenStoredTarget(const Uptane::Target &target) {
```

Comment:
While we're at it, can we change the return type to something like std::ofstream? This is an external API, and IMO we should try to expose less internal/implementation-specific details here.

Reply:
Yes, probably a good idea since we don't store the files in database, so the abstraction is not needed now.

I would leave that for another task/PR though.